internal/elasticsearch/index/index.go

```diff
@@ -307,6 +308,12 @@ func ResourceIndex() *schema.Resource {
 			Description: "Time to delay the allocation of replica shards which become unassigned because a node has left, in time units, e.g. `10s`",
 			Optional:    true,
 		},
+		"mapping_coerce": {
+			Type:        schema.TypeBool,
+			Description: "Set index level coercion setting that is applied to all mapping types. Defaults to `true`.",
+			Optional:    true,
+			Default:     true,
```

Comment:
```suggestion
			Default:     false,
```

ES defaults to false and this would attempt to update the setting on any existing managed indices on upgrade. This setting is also static, so the attempted change would cause TF to fail to apply the change.

Reply:
Oh I thought it's true by checking numeric field, but it looks like global default is `false` but for some types, default is `true`.  Then I think we shouldn't have default value here, so let me remove it🙏

https://github.com/elastic/elasticsearch/blob/006e2acee33fc535578fbbb271f78fd77331d909/server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java#L79

https://github.com/elastic/elasticsearch/blob/006e2acee33fc535578fbbb271f78fd77331d909/server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java#L79